x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/rule_status_dropdown.tsx

```diff
@@ -60,7 +60,6 @@ export const RuleStatusDropdown: React.FunctionComponent<ComponentOpts> = ({
   hideSnoozeOption = false,
   direction = 'column',
 }: ComponentOpts) => {
-  const [isEnabled, setIsEnabled] = useState<boolean>(rule.enabled);
   const [isSnoozed, setIsSnoozed] = useState<boolean>(!hideSnoozeOption && isRuleSnoozed(rule));
```

Comment:
Wondering if `isSnoozed ` could be also removed from useEffect and handled same way as `isEnabled`?

Reply:
I agree with Janki! We can derive the state from the props when the component is rendered.

Reply:
Yep, I quickly went through the `setIsSnoozed` usages and concluded it wasn't necessary since the data flow is unidirectional from `rule`, but a `useMemo` would be much better at handling this, commit coming in a few minutes 🙂

Reply:
Not even a `useMemo` 🙂. `const isSnoozed = !hideSnoozeOption && isRuleSnoozed(rule)` is more than enough!